flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/BatchTableEnvironment.scala

```diff
@@ -228,15 +229,24 @@ abstract class BatchTableEnvironment(
     }
 
     // 3. optimize the logical Flink plan
-    val optRuleSet = getOptRuleSet
-    val flinkOutputProps = relNode.getTraitSet.replace(DataSetConvention.INSTANCE).simplify()
-    val optimizedPlan = if (optRuleSet.iterator().hasNext) {
-      runVolcanoPlanner(optRuleSet, normalizedPlan, flinkOutputProps)
+    val logicalOptRuleSet = getLogicalOptRuleSet
+    val logicalOutputProps = relNode.getTraitSet.replace(FlinkConventions.LOGICAL).simplify()
+    val logicalPlan = if (logicalOptRuleSet.iterator().hasNext) {
+      runVolcanoPlanner(logicalOptRuleSet, normalizedPlan, logicalOutputProps)
```

Comment:
Yes, i will first test how HepPlanner works in some test and product environments and then decide whether or how we change it to rule-based planner. So this will be my follow up task.